### ISSUE TITLE
after sql syntax error occurs, DBeaver will not execute a query until I disconnect and reconnect

### ISSUE THREAD

**Issue by etbres** [dbeaver/dbeaver#2782]
1) Wrote psql query which had syntax error
2) DBeaver threw an error exception
3) I fixed the error and re-executed query
4) DBeaver threw transaction error and does not continue unless I disconnect and reconnect DB connection

PLEASE ADVISE!

Thank you!!

**Comment by serge-rider** [comment]
I assume this is PostgreSQL.
You can:
- Switch to auto-commit mode
- Manually commit or rollback after error
- Enable "Rollback on error" option on Result Sets preference page.

**Comment by etbres** [comment]
Thank you so much!!

**Comment by AndrewKhitrin** [comment]
Sorry. I'll add a little bit important note.

When you use 'Rollback on error' option or rollback manually after error message you rollback ALL CHANGES since last transaction start (commit or session open).

If you use Postgres for prevent session "incorrect" state after syntax or other error you need to use Postgres JDBC Driver support special option autosave : 
.....................
autosave = String

Specifies what the driver should do if a query fails. In autosave=always mode, JDBC driver sets a savepoint before each query, and rolls back to that savepoint in case of failure. In autosave=never mode (default), no savepoint dance is made ever. In autosave=conservative mode, savepoint is set for each query, however the rollback is done only for rare cases like 'cached statement cannot change return type' or 'statement XXX is not valid' so JDBC driver rollsback and retries

The default is 'never'
.............................

In other words if autosave=always driver will rollback only LAST statment.

Hope this helps.
Andrew.

**Comment by hheimbuerger** [comment]
Seems like that option moved to `Database -> Error Handle` in DBeaver 5.0:

![dbeaver_2018-06-01_11-47-59](https://user-images.githubusercontent.com/159414/40834742-efe536ae-6591-11e8-83c1-bacc4741edea.png)
